### PR TITLE
Migrate from Less to Sass

### PR DESCRIPTION
**Pull request by kopijunkie**
# Task
Migrate style code base from Less to Sass. Motivations:
* to be more consistent with rest of company
* will allow us to use Design System v2
* plus other dev benefits

## Checklist
_put `n/a` if item is not relevant to PR changes_

- [x] Has the CHANGELOG been updated?
- [ ] Has the README been updated?
- [ ] Has the CONTRIBUTING doc been updated?
- [ ] Has the RELEASE_GUIDELINES been updated?
- [ ] Has the MIGRATION doc been updated for any MAJOR breaking changes?
- [ ] Has the MIGRATION doc been updated for any MINOR breaking changes, including any translation strings or keys changes?
- [ ] Have any new automated tests been implemented or the existing ones changed?
- [ ] Have any new manual tests been written down or the existing ones changed?
- [ ] Have any new strings been translated or the existing ones changed?

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1072-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1072-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1072-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1072-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1072-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1072-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1072-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1072-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1072-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1072-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1072-pr-onfido-sdk-ui-onfido.surge.sh

File: webpack.config.babel.js [hunk 1]
```diff
@@ -55,23 +54,23 @@ const baseStyleLoaders = (modules, withSourceMap) => [
     loader: 'postcss-loader',
     options: {
       plugins: () => [
-        customMedia(),
+        customMedia({
+          importFrom: `${__dirname}/src/components/Theme/custom-media.css`
```
Comment (stefaniacardenas) [inline webpack.config.babel.js:58]: 👍

File: src/components/Overlay/style.scss [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  $hole-width: 264*$unit;
+  $hole-height: 328*$unit;
+  $offset-y: 105*$unit;
+
+  background-size: ($overlay-width * $scale) ($overlay-height * $scale);
+  background-position: 50% calc(#{($hole-height - $overlay-height) * $scale / 2 + $offset-y});
+}
+
+.face {
+  @include position-hole(1);
```
Comment (stefaniacardenas) [inline src/components/Overlay/style.scss:69]: What's the difference between `@include` and `@extend`?

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1072-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1072-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1072-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Overlay/style.scss (hunk 2 above)
Comment (kopijunkie) [reply src/components/Overlay/style.scss:69]: Good question. `@include` is generally used in Sass for mixins and as you can pass in a parameter while with `@extend` you also can't pass in parameters.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1072-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1072-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1072-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1072-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1072-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1072-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1072-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1072-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1072-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1072-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1072-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1072-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1072-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1072-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1072-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1072-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1072-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by bartzet** [comment]
1. Go to https://1072-pr-onfido-sdk-ui-onfido.surge.sh/?poa=true
2. Tap on 'Verify Identity'
3. Tap on 'Start Verification'
4. Tap on 'Bank or building society statement'
5. Tap on 'Continue'
6. Scroll page to the left and right
7. Observe the 'Take Photo' button

image

**Comment by bartzet** [comment]
Some issues with POA. Looks fine on development 
image

**Comment by kopijunkie** [comment]
> Some issues with POA. Looks fine on development
> image

Have raised a JIRA ticket for this issue as it is a production bug. So will be fixed separately from migration work.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1072-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1072-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by kopijunkie** [comment]
> 1. Go to https://1072-pr-onfido-sdk-ui-onfido.surge.sh/?poa=true
> 2. Tap on 'Verify Identity'
> 3. Tap on 'Start Verification'
> 4. Tap on 'Bank or building society statement'
> 5. Tap on 'Continue'
> 6. Scroll page to the left and right
> 7. Observe the 'Take Photo' button
> 
> Firefox mobile browser
> 
> image

@bartzet double checking on this, we were not able to reproduce this issue on our end with Firefox on Android and iOS, right?

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1072-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by bartzet** [comment]
_@bartzet double checking on this, we were not able to reproduce this issue on our end with Firefox on Android and iOS, right?_

Correct.

**Pull request merged**